### DISCUSSION TITLE
Distinguishing generated code

### DISCUSSION THREAD

**Discussion by isc-tleavitt** [intersystems-community/vscode-objectscript#987] [General]
Is there anything in VSCode to distinguish generated code (i.e., that you shouldn't edit because something else compiling will blow away your changes)? In Studio there's a different background; not sure what this would look like in VSCode.

**Comment by daimor** [comment]
VSCode shows it as read-only, is not enough?
<img width="425" alt="image" src="https://user-images.githubusercontent.com/1212251/192354980-36838e2f-dab6-427f-94c7-767baa229d1d.png">

**Reply by isc-bspead** [reply]
But these are not necessarily read-only (not sure why your .int is readonly? we often edit these to do temp changes that we can blow away with a recompile).
Readonly and Generated would be two different (but both very important) indicators

**Reply by daimor** [reply]
read-only is the default behavior with client-side code editing. There are a few options, that make them editable.

**Reply by isc-bspead** [reply]
since we're discussing crossing over from Studio, we're discussing server-side editing use-cases (in concert with server-side source control).  For Server-side editing, editing .INTs are supported correct?  

We're often interacting with generated server-side code (the specific example that came up today was looking at .disp classes generated by importing a .spec file for REST API generation.

**Comment by daimor** [comment]
But, actually, yes, I'm sure there is a way, to change the background of the code for any Generated code

**Reply by isc-bspead** [reply]
so there is a setting which will make all generated code show up with a different background?

**Reply by daimor** [reply]
Not, requires implementation

**Comment by daimor** [comment]
There is no special way from InterSystems to do this exact thing, but possible to play with custom colorizations
Not so beautiful, but can solve it 
![ezgif com-gif-maker (8)](https://user-images.githubusercontent.com/1212251/192515672-d2598325-0025-46e0-b2e2-bf00ad0f22a7.gif)

What do think?

**Reply by daimor** [reply]
colors have to be specified, at this moment it changes `editor.background` only. 
enabled = false by default. 
```
  "objectscript.generatedCodeBackground": {
    "enabled": true,
    "light": "#ddd",
    "dark": "#444",
  }
```

**Reply by daimor** [reply]
contrast and contrast light themes also support, but no idea, which colors to choose
